automation/publish.sh

```diff
@@ -6,9 +6,40 @@
 # IMAGE_REPO
 # To run it just do proper docker login and automation/publish.sh
 
+image_registry=${IMAGE_REGISTRY:-quay.io}
+image_repo=${IMAGE_REPO:-nmstate}
+branch=${PULL_BASE_REF:-$(git rev-parse --abbrev-ref HEAD)}
+docs_source_branch=master
+docs_container=${image_registry}/${image_repo}/kubernetes-nmstate-docs:$docs_source_branch
+
+# Publish kubernets-nmstate containers
 source automation/check-patch.setup.sh
-cd ${TMP_PROJECT_PATH}
-make \
-    IMAGE_REGISTRY=${IMAGE_REGISTRY:-quay.io}  \
-    IMAGE_REPO=${IMAGE_REPO:-nmstate} \
-    push-handler
+(
+    cd ${TMP_PROJECT_PATH}
+    make \
+        IMAGE_REGISTRY=${image_registry}  \
+        IMAGE_REPO=${image_repo} \
+        push-handler
+)
+
+if [ "$branch" != $docs_source_branch ]; then
+    exit 0
+fi
+
+# Update kubernetes-nmstate documentation container
+docker build docs --build-arg BRANCH=$docs_source_branch -t $docs_container
+docker push $docs_container
```

Comment:
Why is it important to push it? Is it just to cache it for users? And if so, is pulling of this container faster than building it locally? Is it worth the effort? My worry with builder images fetched from the remote registry is that if you need to change something in them it can be quite annoying.

Won't fight against this though.

Reply:
In this case all the things are done in the Makefile so the image is kind of transparent, in fact is more of a CI thing.

We need the image to cache ruby deps, for Jekyll there are some of them that involve compile native code and takes some time.